### ISSUE TITLE
Suggestion command

### ISSUE THREAD

**Issue by Tais993** [Together-Java/TJ-Bot#408]
## See the "Suggestion command" thread for more info

**Is your feature request related to a problem? Please describe.**
Vote manipulation by blocking people, so they can't vote (react). Or by blocking the bot, so the bot can't react on your messages
Also, low effort suggestions, and possible "preferences" because a well known person made the suggestion.

**Describe the solution you'd like**
A command, which creates the suggestion for you based on a filled in "form"

Example layout might be:

> Short description of suggestion:
> ....
> Motivation
> ....
> Drawbacks
> ....
> Benefits
> .... 
> Additional context
> ....

Based on this information, the bot creates an readable embed in an effort to improve the quality of the suggestion.

Whenever the bot mentions the original author of the suggestion can be discussed, not sure or we should / whatever.

**Additional context**
Blocked be #364 (#363) and #382 (#368)

**Comment by Zabuzard** [comment]
unblocked. the mentioned issues are done

**Comment by MaiTheLord** [comment]
I would like to try to implement this, can I assign myself?

**Comment by Zabuzard** [comment]
Feel free, thanks. Id suggest to make this with modals.

**Comment by Tais993** [comment]
[see the wiki](https://github.com/Together-Java/TJ-Bot/wiki/Create-and-use-modals)

**Comment by Taz03** [comment]
what about we only keep 2 fields?? Title and description

**Comment by Tais993** [comment]
Yes but no, making suggestions more "formal" will result in suggestions being easier to implement, and easier to discuss.

We can make drawbacks, benefits and additional context optional, I'd believe it would still result in more people creating a better suggestion.

**Comment by Taz03** [comment]
should we still keep the ability to create a normal suggestion?? i.e. sending a normal message and the bot will add reactions to it

**Comment by Zabuzard** [comment]
im not convinced that a suggest command is helpful tbh. the advantages are minimal. and they increase the entry barrier.

so we have to expect getting a tad less suggestions. and for what benefit? vote manipulation. which never happened on our server yet.

**Comment by Tais993** [comment]
The whole purpose is to maintain high quality of suggestions, which results in the likely-hood of them getting implemented drastically improving. 

just a title & description modal is pointless, only makes it harder.

So if we have a modal design which makes it easy for the suggester to create a high quality suggestion, it's mostly beneficial.

My suggested design is very basic, and applies for almost all suggestions. While it's still less good than someone doing a proper job, it results in the average drastically improving, at least I believe so.


This is far from a priority, but I don't want to sign this off already, we could have a test where we compare what users think, in the future.

**Comment by Zabuzard** [comment]
its just that in order to make this work properly, u would need to also create either sth like the `ImplicitAskListener` (but then lose the advantage of the modal layout), or delete direct messages and having users being mad.

u cant tell discord to only allow slash commands, but no other messages.

and to have the explanation of how this channel works always visible, u need 2 channels. one for posting (which gets cleared again) and one were the suggestions are then displayed. essentially what we have today with `ask_here` and `active_questions`.

i just dont see this working out in a good way. most people are not familiar with slash commands and if u dont make it super obvious, they will fail for it. the amount of people posting questions without `/ask` is also insanely high.

**Comment by Tais993** [comment]
Feel like you're overthinking this drastically.

We pin a message (that resends every message) which tells you to use `/suggestion` in #bots

That's it, nothing more, nothing less.

**Comment by Zabuzard** [comment]
good idea. i think this is a good fusion of both systems.

we can keep it like it works today, but additionally add the slash command on top, so that both synergizes. together with the thing reposting an explanation so that its on the bottom.

im fine with that idea. but i personally dont really give it a high prio. but its legit.

**Comment by Taz03** [comment]
what fields should there be on the form??

i vote for title(short description), description and benefits

**Comment by ankitsmt211** [comment]
Great idea about making it anonymous, after a certain "threshold" is crossed it should be open for further discussions. I'm assuming moderators will still be able to keep track of things, just to keep it regulated. Maybe we can have each suggestion to have a 24 hour period for votes, after that it will either be "disregarded" or marked as "need more information" or "open for discussion" based on votes. Possibily another suggestion can be raised if "need more info" is somewhat on the higher side.

**Comment by ankitsmt211** [comment]
To add to previous comment, about the reactions. A user is more likely to agree to something that most people have already agreed to. It's like the "herd" mentality, at least a normal user who's not really looking at the suggestions objectively before showing interest. To avoid this, a suggestion during the voting period the count of votes or reaction shouldn't be visible to anyone but maybe mods. That way truly making the vote unbiased/motivated in any form at least to the limit we can.
